spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/Flight.java

```diff
@@ -0,0 +1,271 @@
+    this.distance = distance;
+    this.delayed = delayed;
+  }
+
+  public Flight(int dofM, int dofW, Double carrier, String tailNum, int flightNum, int originId, String origin,
+                int destId, String dest, double scheduleDepTime, Double deptime, Double depDelayMins,
+                Double scheduledArrTime, Double arrTime, Double arrDelay, Double elapsedTime, int distance) {
+    this.dofM = dofM;
+    this.dofW = dofW;
+    this.carrier = carrier;
+    this.tailNum = tailNum;
+    this.flightNum = flightNum;
+    this.originId = originId;
+    this.origin = origin;
+    this.destId = destId;
+    this.dest = dest;
+    this.scheduleDepTime = scheduleDepTime;
+    this.deptime = deptime;
+    this.depDelayMins = depDelayMins;
+    this.scheduledArrTime = scheduledArrTime;
+    this.arrTime = arrTime;
+    this.arrDelay = arrDelay;
+    this.elapsedTime = elapsedTime;
+    this.distance = distance;
+    this.delayed = null;
+  }
+
+  public static Flight fromStructuredRecord(StructuredRecord structuredRecord) {
+    return new Flight((Integer) structuredRecord.get("dofM"),
+                      (Integer) structuredRecord.get("dofW"),
+                      (Double) structuredRecord.get("carrier"),
+                      (String) structuredRecord.get("tailNum"),
+                      (Integer) structuredRecord.get("flightNum"),
+                      (Integer) structuredRecord.get("originId"),
+                      (String) structuredRecord.get("origin"),
+                      (Integer) structuredRecord.get("destId"),
+                      (String) structuredRecord.get("dest"),
+                      (Double) structuredRecord.get("scheduleDepTime"),
+                      (Double) structuredRecord.get("deptime"),
+                      (Double) structuredRecord.get("depDelayMins"),
+                      (Double) structuredRecord.get("scheduledArrTime"),
+                      (Double) structuredRecord.get("arrTime"),
+                      (Double) structuredRecord.get("arrDelay"),
+                      (Double) structuredRecord.get("elapsedTime"),
+                      (Integer) structuredRecord.get("distance"),
+                      (Double) structuredRecord.get("delayed"));
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Flight)) {
+      return false;
+    }
+
+    Flight flight = (Flight) o;
+
+    if (dofM != flight.dofM) {
+      return false;
+    }
+    if (dofW != flight.dofW) {
+      return false;
+    }
+    if (flightNum != flight.flightNum) {
+      return false;
+    }
+    if (originId != flight.originId) {
+      return false;
+    }
+    if (destId != flight.destId) {
+      return false;
+    }
+    if (Double.compare(flight.scheduleDepTime, scheduleDepTime) != 0) {
+      return false;
+    }
+    if (distance != flight.distance) {
+      return false;
+    }
+    if (!carrier.equals(flight.carrier)) {
```

Comment:
this can throw an exception if carrier is null. You can use java.util.Objects.equals() and it will handle nulls for you. For consistent style with other classes, you can do something like:

```
return dofM == flight.dofM &&
  dofW == flight.dofW && 
  Objects.equals(carrier, flight.carrier) && 
  ...
```